src/runtime/FunctionObject.cpp

```diff
@@ -270,54 +270,49 @@ NEVER_INLINE void FunctionObject::generateBytecodeBlock(ExecutionState& state)
     currentCodeSizeTotal += m_codeBlock->m_byteCodeBlock->memoryAllocatedSize();
 }
 
-Value FunctionObject::callSlowCase(ExecutionState& state, const Value& callee, const Value& receiver, const size_t argc, Value* argv, bool isNewExpression)
-{
-    if (LIKELY(callee.isObject())) {
-        if (LIKELY(callee.asPointerValue()->isFunctionObject()))
-            return callee.asFunction()->processCall(state, receiver, argc, argv, isNewExpression);
-#if ESCARGOT_ENABLE_PROXY_REFLECT
-        else if (callee.asPointerValue()->isProxyObject() && callee.asPointerValue()->isCallable())
-            return callee.asPointerValue()->asProxyObject()->call(state, callee, receiver, argc, argv);
-#endif
-    }
-    ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, errorMessage_Call_NotFunction);
-    return Value();
-}
-
-Object* FunctionObject::newInstance(ExecutionState& state, const size_t argc, Value* argv)
+// https://www.ecma-international.org/ecma-262/6.0/#sec-ecmascript-function-objects-construct-argumentslist-newtarget
+Object* FunctionObject::construct(ExecutionState& state, const size_t argc, Value* argv, const Value& newTarget)
```

Comment:
Check pointer-type parameter / check before deferencing
ASSERT(argv != nullptr)

Reply:
`argv` could be a null pointer, so I added `NULLABLE` keyword to notify it.